dev/infra/Dockerfile

```diff
@@ -64,7 +64,7 @@ RUN Rscript -e "devtools::install_version('roxygen2', version='7.2.0', repos='ht
 # See more in SPARK-39735
 ENV R_LIBS_SITE "/usr/local/lib/R/site-library:${R_LIBS_SITE}:/usr/lib/R/library"
 
-RUN pypy3 -m pip install numpy 'pandas<=1.5.3' scipy coverage matplotlib
+RUN pypy3 -m pip install numpy pyarrow 'pandas<=1.5.3' scipy coverage matplotlib
```

Comment:
This is only needed to test Python code in docstrings. However, installing pyarror in pypy is not easy.